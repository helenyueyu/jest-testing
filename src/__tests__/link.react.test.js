import React from 'react'
import Link from '../components/link.react'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<Link page="http://www.facebook.com">Facebook</Link>)
  .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders as an anchor when no page is set', () => {
  const tree = renderer.create(<Link>Facebook</Link>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('properly escapes quotes', () => {
  const tree = renderer
    .create(<Link>{"\"Facebook\" \\'is\\' awesome'"}</Link>).toJSON()
    expect(tree).toMatchSnapshot()
})
