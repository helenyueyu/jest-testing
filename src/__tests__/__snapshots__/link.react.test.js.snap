// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly escapes quotes 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  "Facebook" \\'is\\' awesome'
</a>
`;

exports[`renders as an anchor when no page is set 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
